### DISCUSSION TITLE
Validating Access Token

### DISCUSSION THREAD

**Discussion by tatlicak** [etsy/open-api#1035] [Q&A]
Hi, 

I am new in Etsy Api v3. I searched documentation for hours but i couldn't found any endpoint for token validation. I wanna control my access token valid or not valid before send request to any endpoint. Is there any other way to validate access token before sending a request to endpoint. Can you help me about this.

Regards

**Comment by Gareth064** [comment]
Hi @tatlicak  welcome to the community.

Please read the following documentation. It explains how authentication works, along with what to do about access tokens etc

https://developers.etsy.com/documentation/essentials/authentication

Once you've read through this, if you have any questions then please come back here and ask.

**Comment by turboturtleD** [comment]
The best way to validate a token is using the tokenScopes endpoint: https://developers.etsy.com/documentation/reference#operation/tokenScopes

You could also use the ping endpoint, but it doesn't require the oAuth token and would not validate an obtained token: https://developers.etsy.com/documentation/reference#operation/ping
